src/process.cr

```diff
@@ -50,15 +50,18 @@ class Process
     Crystal::System::Process.signal(pid, signal.value)
   end
 
+  alias Interrupt = Crystal::System::Process::Interrupt
+
   # Installs *handler* as the new handler for interrupt requests. Removes any
   # previously set interrupt handler.
   #
   # The handler is executed on a fresh fiber every time an interrupt occurs.
   #
-  # * On Unix-like systems, this traps `SIGINT`.
-  # * On Windows, this captures <kbd>Ctrl</kbd> + <kbd>C</kbd> and
-  #   <kbd>Ctrl</kbd> + <kbd>Break</kbd> signals sent to a console application.
-  def self.on_interrupt(&handler : ->) : Nil
+  # * On Unix-like systems, this traps `SIGINT`, `SIGHUP`, `SIGTERM`.
+  # * On Windows, this captures <kbd>Ctrl</kbd> + <kbd>C</kbd>,
+  #   <kbd>Ctrl</kbd> + <kbd>Break</kbd>, terminal close, windows logoff
+  #   and shutdown signals sent to a console application.
+  def self.on_interrupt(&handler : Interrupt ->) : Nil
```

Comment:
I am not sure if there needs to be a single handler physically shared between the different kinds of interrupts. If we have gone this far, we could also have defined something like `on_interrupt(kind : Interrupt, handler : ->)`, and I am certain it is possible to make Win32's `SetConsoleCtrlHandler` work with such a Crystal API.

Reply:
I think if we were to do this we should probably revert to having a separate enum again.
I don't see much value in splitting up the interrupt types, just makes it harder for the programmer to handle.

As far as I can tell the main benefit of these differing interrupts is so the OS can signal why the app has been requested to shutdown. The user can check the enum for the reason and perform different actions if they want.

However in 99.999% of cases we just want to handle exiting cleanly and I don't see a reason to add complexity to it